component/pdp/capabilities.go

```diff
@@ -0,0 +1,112 @@
+package pdp
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+)
+
+//go:embed capabilities/*.json
+var FS embed.FS
+
+func readCapability(name string) (fhir.CapabilityStatement, error) {
+	fileName := fmt.Sprintf("capabilities/%s.json", name)
+	data, err := FS.ReadFile(fileName)
+	if err != nil {
+		return fhir.CapabilityStatement{}, err
+	}
+
+	var capability fhir.CapabilityStatement
+	if err := json.Unmarshal(data, &capability); err != nil {
+		log.Warn().Err(err).Msg(fmt.Sprintf("unable to read JSON in %s", fileName))
+		return fhir.CapabilityStatement{}, err
+	}
+
+	return capability, nil
+}
+
+func capabilityForScope(scope string) (fhir.CapabilityStatement, bool) {
+	switch scope {
+	// FUTURE: Should be made configurable or packaged up with some policy
+	case "mcsd_update":
+		capa, err := readCapability("nl-gf-admin-directory-update-client")
+		return capa, err == nil
+	case "patient_example":
+		capa, err := readCapability("patient-example")
+		return capa, err == nil
+	default:
+		return fhir.CapabilityStatement{}, false
+	}
+}
+
+func EvalCapabilityPolicy(input MainPolicyInput) PolicyResult {
+	out := PolicyResult{
+		Allow: false,
+	}
+
+	if input.Scope == "" {
```

Comment:
does this function need to be exported? Because you already did this in the calling function, so if you make this private, you don't need to re-check this?